WEB-INF/classes/gov/noaa/pfel/erddap/LoadDatasets.java

```diff
@@ -861,441 +1012,237 @@ public void run() {
-                //unload them
-                Iterator it = orphanIDSet.iterator();
-                while (it.hasNext()) 
-                    tryToUnload(erddap, (String)it.next(), changedDatasetIDs, false);  //needToUpdateLucene
-                updateLucene(erddap, changedDatasetIDs);
-
-                String msg = String2.ERROR + ": n Orphan Datasets removed (datasets in ERDDAP but not in datasets.xml) = " + orphanIDSet.size() + "\n" +
-                    "    " + 
-                    String2.noLongLinesAtSpace(String2.toCSSVString(orphanIDSet), 100, "    ") + 
-                    "(end)\n";
-                errorsDuringMajorReload += msg;
-                String2.log(msg);
-                EDStatic.email(EDStatic.emailEverythingToCsv, "Orphan Datasets Removed", msg);
-            }
-
-            EDStatic.nGridDatasets = erddap.gridDatasetHashMap.size();
-            EDStatic.nTableDatasets = erddap.tableDatasetHashMap.size();
-
-            //*** print lots of useful information
-            long loadDatasetsTime = System.currentTimeMillis() - startTime;
-            String cDateTimeLocal = Calendar2.getCurrentISODateTimeStringLocalTZ();
-            String2.log("\n" + String2.makeString('*', 80) + "\n" + 
-                "LoadDatasets.run finished at " + cDateTimeLocal + 
-                "  TOTAL TIME=" + loadDatasetsTime + "ms\n" +
-                "  nGridDatasets active=" + EDStatic.nGridDatasets + 
-                    " change=" + (EDStatic.nGridDatasets - oldNGrid) + "\n" +
-                "  nTableDatasets active=" + EDStatic.nTableDatasets + 
-                    " change=" + (EDStatic.nTableDatasets - oldNTable) + "\n" +
-                "  nDatasets in datasets.xml=" + nDatasets + " (nTry=" + nTry + ")\n" +
-                "  nUsers=" + tUserHashMap.size());
-
-            //minorLoad?
-            if (!majorLoad) {
-                String2.distributeTime(loadDatasetsTime, EDStatic.minorLoadDatasetsDistribution24);
-                String2.distributeTime(loadDatasetsTime, EDStatic.minorLoadDatasetsDistributionTotal);
-                String2.log(datasetsThatFailedToLoad);
+        } catch (Throwable t) {
+            String subject = String2.ERROR + " while processing " +
+                    (xmlReader == null? "" : "line #" + xmlReader.lineNumber() + " ") +
+                    "datasets.xml";
+            if (!isInterrupted()) {
+                EDStatic.errorsDuringMajorReload  =
+                        subject + ": see log.txt for details.\n";  //swap into place
+                String content = MustBe.throwableToString(t);
+                unexpectedError = subject + ": " + content;
+                String2.log(unexpectedError);
+                EDStatic.email(EDStatic.emailEverythingToCsv, subject, content);
             }
+        } finally {
+            if (xmlReader != null)
+                try {xmlReader.close();} catch (Exception e) {}
+            EDStatic.suggestAddFillValueCSV.setLength(0);
```

Comment:
I believe `EDStatic.suggestAddFillValueCSV` is used as part of the `handleAfva`. I think the call `EDStatic.suggestAddFillValueCSV.setLength(0);` is unrelated to the xml parsing function and so should be in the main run.